### PR TITLE
Chore: upgrade libraries and fix security issues

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Upgrade go libraries. Part of the effort to resolve security issues.

#496

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,11 +23,7 @@ func main() {
 	opts := &plugin.ServeOpts{ProviderFunc: env0.Provider(version)}
 
 	if debugMode {
-		err := plugin.Debug(context.Background(), "registry.terraform.io/env0/env0", opts)
-		if err != nil {
-			log.Fatal(err.Error())
-		}
-		return
+		opts.Debug = true
```
Comment (razbensimon) [inline main.go:26]: can you explain more about this change compared to what we had before?

File: utils/test_reporter.go [hunk 2]
```diff
@@ -86,3 +86,7 @@ func (r *TestReporter) Errorf(format string, args ...interface{}) {
 	r.T.Helper()
 	r.T.Errorf(format, args...)
 }
+
+func (r *TestReporter) Cleanup(cb func()) {
+	r.T.Helper()
+}
```
Comment (razbensimon) [inline utils/test_reporter.go:92]: no call?

**Review by razbensimon — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (TomerHeber) [reply main.go:26]: The old code has been deprecated.
I reviewed the documentation, and this is what they suggest doing instead.

File: utils/test_reporter.go (hunk 2 above)
Comment (razbensimon) [reply utils/test_reporter.go:92]: > I'll add it.
I don't say we need it, just asking how we add a function that we don't use. what is the reason for this one

**Review by razbensimon — APPROVED** [review]

File: utils/test_reporter.go (hunk 2 above)
Comment (TomerHeber) [reply utils/test_reporter.go]: It's required because it must implement the interface. This is a new function added to the interface.

**Pull request merged**